sign_mac.sh

```diff
@@ -6,28 +6,49 @@ set -x
 
 for DMG in obj/dist/*.dmg
 do
   echo "Processing $DMG..."
   rm -f -rf $PKG_DIR
   mkdir $PKG_DIR
-  hdiutil attach -nobrowse $DMG
-  cp -r /Volumes/cliqz/*.app $PKG_DIR
-  xattr -rc $PKG_DIR/
-  hdiutil detach /Volumes/cliqz
-  for app in $PKG_DIR/*.app
-  do
-    codesign -s $MAC_CERT_NAME --force --deep $app
-
-    # copy back to dist folder a signed app (for generating an update package(s) later)
-    if [[ $DMG == *"de.mac.dmg"* ]]; then
-      cp -r $app obj/dist/l10n-stage/cliqz
-    fi
-    if [[ $DMG == *"en-US.mac.dmg"* ]]; then
-      cp -r $app obj/dist/cliqz
-    fi
-  done
+  mozilla-release/build/package/mac_osx/unpack-diskimage $DMG /Volumes/Cliqz $PKG_DIR
+  xattr -cr $PKG_DIR/Cliqz.app
+  xattr -cr $PKG_DIR/Cliqz.app/Contents/MacOS/plugin-container.app
+  xattr -cr $PKG_DIR/Cliqz.app/Contents/MacOS/crashreporter.app
+  xattr -cr $PKG_DIR/Cliqz.app/Contents/MacOS/updater.app
+
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/plugin-container.app
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/pingsender
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libfreebl3.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/cliqz-bin
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/liblgpllibs.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libplugin_child_interpose.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libsoftokn3.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/XUL
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libmozavutil.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libmozglue.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libmozavcodec.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libnssckbi.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/libnss3.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/minidump-analyzer
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/crashreporter.app
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/crashreporter.app/Contents/MacOS/crashreporter
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/updater.app
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/MacOS/updater.app/Contents/MacOS/org.mozilla.updater
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/Library/LaunchServices/org.mozilla.updater
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app/Contents/Resources/gmp-clearkey/0.1/libclearkey.dylib
+  codesign "${SIGN_ARGS[@]}" $PKG_DIR/Cliqz.app
+  codesign -vvv --deep --strict $PKG_DIR/Cliqz.app
```

Comment:
Isn't `--deep` already codesigning everything inside?

Reply:
Unfortunately, not. Also please check args which we use for signing. I just repeat all steps which Mozilla's do for Firefox. All this step is necessary for proper notarization afterward